pkg/operator/mon/health.go

```diff
@@ -77,9 +78,26 @@ func (c *Cluster) checkHealth() error {
 		inQuorum := monInQuorum(mon, status.Quorum)
 		if inQuorum {
 			logger.Debugf("mon %s found in quorum", mon.Name)
+			// delete the "timeout" for a mon if the pod is in quorum again
+			if _, ok := c.monTimeoutList[mon.Name]; ok {
+				delete(c.monTimeoutList, mon.Name)
+			}
 		} else {
 			logger.Warningf("mon %s NOT found in quorum. %+v", mon.Name, status)
 
+			// If not yet set, add the current time, for the timeout
+			// calculation, to the list
+			if _, ok := c.monTimeoutList[mon.Name]; !ok {
+				c.monTimeoutList[mon.Name] = time.Now()
+			}
+
+			// when the timeout for the mon has been reached, continue to the
+			// normal failover/delete mon pod part of the code
+			if time.Since(c.monTimeoutList[mon.Name]) > MonOutTimeout {
```

Comment:
looks like the mon still didn't failover in the test. Were you able to test the failover locally in a cluster? For example, delete a mon replicaset and see if the operator fails the mon over. If that doesn't work locally, I wonder if this timeout condition is not right

Reply:
I should have fixed the issue now. It was a nil panic that caused the failover to not happen.
Also I ran into some issues that I added to the commit.